src/store/index.ts

```diff
@@ -80,41 +94,98 @@ export default createStore<State>({
         payload: messageData.ciphertext
       });
     },
-    async uploadFile({ dispatch }, request: UploadRequest) {
+    async uploadFile({ dispatch, getters }, request: UploadRequest) {
       const res = await axios.post<UploadResponse>(
         "https://rtc-data.registree.io/data/uploads",
         request
       );
       if (res.status !== 200) {
         return "error";
       }
-      await dispatch("decryptUploadResponse", res.data);
-    },
-    async decryptUploadResponse(
-      { state, dispatch, getters },
-      response: UploadResponse
-    ) {
+
+      const { nonce, ciphertext } = res.data;
       const enclavePubKey = getters.enclavePublicKey;
-      if (!enclavePubKey || !state.ourSecretKey) {
+      const ourSecretKey = getters.ourSecretKey;
+
+      const msg = decryptMessage(
+        ciphertext,
+        nonce,
+        enclavePubKey,
+        ourSecretKey
+      );
+
+      if (!msg) {
+        // TODO: Error Handling
         return "error";
       }
+
+      await dispatch("parseUploadMessage", msg);
+    },
+    async postAccessForm({ commit, getters }, request: UploadRequest) {
+      const res = await axios.post<UploadResponse>(
+        "https://rtc-data.registree.io/auth/tokens",
+        request
+      );
+      if (res.status !== 200) {
+        return "error";
+      }
+
+      const { nonce, ciphertext } = res.data;
+      const enclavePubKey = getters.enclavePublicKey;
+      const ourSecretKey = getters.ourSecretKey;
+
       const msg = decryptMessage(
-        response.ciphertext,
-        response.nonce,
+        ciphertext,
+        nonce,
         enclavePubKey,
-        state.ourSecretKey
+        ourSecretKey
       );
+
       if (!msg) {
         // TODO: Error Handling
         return "error";
       }
-      await dispatch("parseUploadMessage", msg);
+
+      //TODO: Open Dialog Box show message
+
+      commit("setDecryptedMsg", msg);
+    },
+    async decryptResponse({ state, getters }, response: UploadResponse) {
+      const enclavePubKey = getters.enclavePublicKey;
+      if (!enclavePubKey || !state.ourSecretKey) {
+        return "error";
+      }
+      return decryptMessage(
+        response.ciphertext,
+        response.nonce,
+        enclavePubKey,
+        state.ourSecretKey
+      );
     },
     async parseUploadMessage({ commit }, message: Uint8Array) {
       console.log("parseUploadMessage:", message);
       const accessKey = btoa(message.slice(0, 24).toString());
       const uuid = message.slice(24);
       commit("saveUploadResult", { accessKey, uuid });
+    },
+    async requestExecutionToken({ commit, dispatch, getters }, data) {
+      const enclavePubKey = getters.enclavePublicKey;
+      if (!enclavePubKey) {
+        return null;
```

Comment:
We might want to error out here rather than just returning. If this is temporary until the endpoints are up, just add a `// TODO`